client.go

```diff
@@ -49,6 +49,7 @@ type QueryParam struct {
 	WantUnicastResponse bool                 // Unicast response desired, as per 5.4 in RFC
 	DisableIPv4         bool                 // Whether to disable usage of IPv4 for MDNS operations. Does not affect discovered addresses.
 	DisableIPv6         bool                 // Whether to disable usage of IPv6 for MDNS operations. Does not affect discovered addresses.
+	Log                 *log.Logger          // Optionally provide a *log.Logger to better manage log output.
```

Comment:
Why `Log` here and `Logger` on the other exported struct field. I think we typically name these sorts of optional config fields `Logger` over in serf & memberlist support libraries.